### ISSUE TITLE
Fix code scanning alert - CVE-2020-8565

### ISSUE THREAD

**Issue by aaronshurley** [carvel-dev/kapp-controller#436]
Tracking issue for:
- [ ] https://github.com/vmware-tanzu/carvel-kapp-controller/security/code-scanning/23

**Comment by joe-kimmel-vmw** [comment]
a few notes trying to figure out what's the deal with this CVE alert: 

details on the CVE from
https://groups.google.com/g/kubernetes-security-discuss/c/vm-HcrFUOCs/m/36utxAM5CwAJ
and
https://github.com/kubernetes/kubernetes/issues/95623

both agree that this is fixed in kubernetes v1.19.6+. (or 19.3+ , depending on which CVE you're checking)
Oh cool - what version of kubernetes does kapp controller use?   Well no, seriously - how would you tell?  I'm guessing the important detail is actually the version of kubernetes running in the cluster and not the version of the k8s libraries, but also I'm hoping that's true because let me tell you about the gates of madness behind our k8s library versions:

There's a bunch of libraries:  some have the version 0.19.15 (so we might think, "well maybe the k8s libraries are 1 major version behind k8s itself?" in which case we should already be past this CVE)  others have the version 0.19.2 (so we might think, "well maybe we're mixing and matching two different versions of kubernetes in our library usage?" in which case we should upgrade to a uniform version both because it's the right thing to do morally and to get past this CVE), and others have versions that are all completely dissimilar to the other versions so might think a being from the dimension of endless suffering has intentionally scrambled all useful information. 
I tried upgrading the libraries that were 0.19.2 to also be 0.19.15 but then `go mod` starts saying things like `github.com/vmware-tanzu/carvel-kapp-controller/pkg/client/clientset/versioned: no matching versions for query "latest"`.  And even if that's true, how could changing from one pinned version to a different pinned version trigger existential crises in resolving "latest" tags?   So like I said, I'm mostly just hoping that what actually matters is the runtime version, which we have no control over anyhow.


In general the other solution (besides "don't run old/unpatched versions of kubernetes") is "don't run in production with a high log level" where note that the logs in question are ordered from low (critical) to high (debug). 

in contrast the logs in kapp-controller seem to be ordered from low (debug) to high (critical) https://github.com/vmware-tanzu/carvel-kapp-controller/blob/develop/vendor/go.uber.org/zap/level.go#L33

It's not immediately clear if these logs are the same or different or if it warrants bumping the default log level in kapp-controller to be higher than info (or lower than info? i already forgot), but again the CVE docs do seem to imply that this is a cluster wide setting and the owner/operator of the cluster is ultimately responsible.

So in conclusion my hope is that this CVE is a no-op for us because I can't tell what action Trivy is trying to imply we should take.

**Comment by joe-kimmel-vmw** [comment]
closing this as we feel the CVE was a false positive, and our version of Kubernetes library (0.19.15) is sufficient.
We also don't think our log levels would be compromising, even in earlier versions of kubernetes.
